utils/ort/src/main/kotlin/OrtAuthenticator.kt

```diff
@@ -86,6 +86,12 @@ class OrtAuthenticator(private val original: Authenticator? = null) : Authentica
             }
 
             RequestorType.SERVER -> {
+                requestingURL.userInfo?.let {
```

Comment:
I'm actually surprised that `getPasswordAuthentication()` even gets called when the credentials are in the URL... but I guess you've verified that this happens in real-life.

Reply:
Do you think it's worth implementing this is a separate `UserInfoAuthenticator` that is added as the first entry to `delegateAuthenticators`?

Reply:
> I'm actually surprised that `getPasswordAuthentication()` even gets called when the credentials are in the URL... but I guess you've verified that this happens in real-life.

 Yes this is pretty strange behavior.

My understanding is, when an `Authenticator` is installed, all requests goes through a check in  the JDK. The latter consists of test request to the server (HEAD ?) to see if it requires authentication. If yes, it goes through the `Authenticator`.
When a request comes in with userinfo,  the latter is stripped before doing this test request, hence the request lands in `Authenticator`.

I could not find anything online to back it up so it's just my fuzzy explanation.

> Do you think it's worth implementing this is a separate `UserInfoAuthenticator` that is added as the first entry to `delegateAuthenticators`?

Good idea I will change my PR.